src/dac.rs

```diff
@@ -0,0 +1,248 @@
+impl Dac {
+    /// Create a new DAC instance
+    pub fn new<P: Pins>(
+        regs: pac::DAC,
+        _pins: P,
+        channel: Channel,
+        bits: DacBits,
+        vref: f32,
+    ) -> Self {
+        Self {
+            regs,
+            channel,
+            bits,
+            vref,
+        }
+    }
+
+    /// Enable the DAC.
+    pub fn enable(&mut self, apb1: &mut APB1) {
+        match self.channel {
+            Channel::One => {
+                apb1.enr().modify(|_, w| w.dac1en().enabled());
+                self.regs.cr.modify(|_, w| w.en1().enabled());
+            }
+            Channel::Two => {
+                apb1.enr().modify(|_, w| w.dac2en().enabled());
+                self.regs.cr.modify(|_, w| w.en2().enabled());
+            }
+        }
+    }
+
+    /// Disable the DAC
+    pub fn disable(&mut self, apb1: &mut APB1) {
+        match self.channel {
+            Channel::One => {
+                self.regs.cr.modify(|_, w| w.en1().disabled());
+                apb1.enr().modify(|_, w| w.dac1en().disabled());
+            }
+            Channel::Two => {
+                self.regs.cr.modify(|_, w| w.en2().disabled());
+                apb1.enr().modify(|_, w| w.dac2en().disabled());
+            }
+        }
+    }
+
+    /// Set the DAC value as an integer.
+    pub fn set_value(&mut self, val: u32) {
+        match self.channel {
+            Channel::One => match self.bits {
+                DacBits::EightR => self.regs.dhr8r1.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveL => self.regs.dhr12l1.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveR => self.regs.dhr12r1.modify(|_, w| unsafe { w.bits(val) }),
+            },
+            Channel::Two => match self.bits {
+                DacBits::EightR => self.regs.dhr8r2.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveL => self.regs.dhr12l2.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveR => self.regs.dhr12r2.modify(|_, w| unsafe { w.bits(val) }),
+            },
+        }
+    }
+
+    /// Set the DAC voltage. `v` is in Volts.
+    pub fn set_voltage(&mut self, volts: f32) {
+        let val = match self.bits {
+            DacBits::EightR => ((volts / self.vref) * 255.) as u32,
+            DacBits::TwelveL => ((volts / self.vref) * 4_095.) as u32,
+            DacBits::TwelveR => ((volts / self.vref) * 4_095.) as u32,
+        };
+
+        self.set_value(val);
+    }
+
+    // Select and activate a trigger. See f303 Reference manual, section 16.5.4.
+    pub fn set_trigger(&mut self, trigger: Trigger) {
+        match self.channel {
+            Channel::One => {
+                self.regs.cr.modify(|_, w| w.ten1().set_bit());
+                self.regs
+                    .cr
+                    .modify(|_, w| unsafe { w.tsel1().bits(trigger.bits()) });
+            }
+            Channel::Two => {
+                self.regs.cr.modify(|_, w| w.ten2().set_bit());
```

Comment:
The `tenX` fields also have `enabled()` setters, which are preferable to `set_bit()`